staff/views/projects.py

```diff
@@ -149,6 +151,39 @@ def post(self, request, *args, **kwargs):
         return redirect(self.project.get_staff_feature_flags_url())
 
 
+@method_decorator(require_role(CoreDeveloper), name="dispatch")
+class ProjectLinkApplication(UpdateView):
+    form_class = ProjectLinkApplicationForm
+    model = Project
+    template_name = "staff/project_link_application.html"
+
+    def form_valid(self, form):
+        application = form.cleaned_data["application"]
+
+        self.object.applications.add(application)
+        self.object.save()
+
+        return redirect(self.object.get_staff_url())
+
+    def get_context_data(self, **kwargs):
+        # get applications that we can consider done, or at least not clearly
+        # not-done
+        ignored_statuses = [
+            Application.Statuses.ONGOING,
+            Application.Statuses.REJECTED,
+        ]
+        applications = (
+            Application.objects.filter(project=None)
+            .exclude(status__in=ignored_statuses)
+            .select_related("created_by")
+            .order_by("-created_at")
+        )
+
+        return super().get_context_data(**kwargs) | {
+            "applications": applications,
+        }
```

Comment:
first time I've seen this syntax in the wild :slightly_smiling_face:

Reply:
It's really cut down the boiler plate when doing simple extensions of the GCBV bits!